packages/router/router.ts

```diff
@@ -1142,10 +1151,10 @@ export function createRouter(init: RouterInit): Router {
     if (matchesToLoad.length === 0 && revalidatingFetchers.length === 0) {
       completeNavigation(location, {
         matches,
-        loaderData: mergeLoaderData(state.loaderData, {}, matches),
+        loaderData: {},
```

Comment:
This is a duplicative call, we can just let the empty object be merged inside `completeNavigation` now